gnomad/assessment/summary_stats.py

```diff
@@ -0,0 +1,131 @@
+from typing import Dict
+
+import hail as hl
+
+from gnomad.utils.filtering import filter_low_conf_regions
+from gnomad.utils.vep import (
+    filter_vep_to_canonical_transcripts,
+    get_worst_consequence_for_summary,
+)
+
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+def freq_criteria(
+    freq_expr: hl.expr.ArrayExpression, index: int = 0
+) -> hl.expr.builders.CaseBuilder:
+    """
+	Returns case statement adding frequency string annotations based on input AC or AF.
+
+	:param freq_expr: Array of structs containing frequency information.
+	:param index: Which index of freq_expr to use for annotation. Default is 0 (all pops calculated on adj genotypes only).
+	:return: Case statement adding string expressions based on input AC or AF.
+	"""
+    return (
+        hl.case(missing_false=True)
+        .when(freq_expr[index].AC == 0, "Not found")
+        .when(freq_expr[index].AC == 1, "Singleton")
+        .when(freq_expr[index].AC == 2, "Doubleton")
+        .when(freq_expr[index].AC <= 5, "AC 3 - 5")
+        .when(freq_expr[index].AF < 1e-4, "AC 6 - 0.01%")
+        .when(freq_expr[index].AF < 1e-3, "0.01% - 0.1%")
+        .when(freq_expr[index].AF < 1e-2, "0.1% - 1%")
+        .when(freq_expr[index].AF < 1e-1, "1% - 10%")
+        .when(freq_expr[index].AF > 0.95, ">95%")
+        .default("10% - 95%")
+    )
+
+
+def get_summary_counts_dict(
+    allele_expr: hl.expr.ArrayExpression,
+    lof_expr: hl.expr.StringExpression,
+    no_lof_flags_expr: hl.expr.BooleanExpression,
+    prefix_str: str = "",
+) -> Dict[str, hl.expr.Int64Expression]:
+    """
+	Returns dictionary containing desired categories and their counts.
+
+	Assumes allele_expr contains only two variants (multi-allelics have been split).
+
+	:param allele_expr: ArrayExpression containing alleles.
+	:param lof_expr: StringExpression containing LOFTEE annotation.
+	:param no_lof_flags_expr: BooleanExpression indicating whether LoF variant has any flags.
+	:param prefix_str: Desired prefix string for category names. Default is empty str.
+	:return: Dict of categories and counts per category.
+	"""
+    return {
+        f"{prefix_str}num_variants": hl.agg.count(),
+        f"{prefix_str}indels": hl.agg.count_where(
+            hl.is_indel(allele_expr[0], allele_expr[1])
+        ),
+        f"{prefix_str}snps": hl.agg.count_where(
+            hl.is_snp(allele_expr[0], allele_expr[1])
+        ),
+        f"{prefix_str}LOF": hl.agg.count_where(hl.is_defined(lof_expr)),
+        f"{prefix_str}pass_loftee": hl.agg.count_where(lof_expr == "HC"),
+        f"{prefix_str}pass_loftee_no_flag": hl.agg.count_where(
+            (lof_expr == "HC") & (no_lof_flags_expr)
+        ),
+        f"{prefix_str}loftee_os": hl.agg.count_where(lof_expr == "OS"),
+        f"{prefix_str}fail_loftee": hl.agg.count_where(lof_expr == "LC"),
+    }
+
+
+def get_summary_per_variant(
```

Comment:
Not sure this is the right name. Is it per variant? It's a summary of stats per bin defined in `freq_criteria` and category defined in `get_summary_counts_dict` right?